crates/iota-core/src/subscription_handler.rs

```diff
@@ -104,10 +107,44 @@ impl SubscriptionHandler {
         }
 
         // serially dispatch event processing to honor events' orders.
-        for event in events.data.clone() {
-            if let Err(e) = self.event_streamer.try_send(event) {
+        for (index, event) in events.data.clone().iter().enumerate() {
+            // Send to existing event streamer
+            if let Err(e) = self.event_streamer.try_send(event.clone()) {
                 error!(error =? e, "Failed to send event to dispatch");
             }
+
+            // Also send to gRPC broadcast channel if available
+            if let Some(ref grpc_event_broadcaster) = self.grpc_event_broadcaster {
```

Comment:
Should `self.grpc_even_broadcaster` use `self.event_streamer`? `self.event_streamer` seems to be designed to handle exactly this case: it only sends events to those subscribers with matching filters. `EventGrpcService` would just duplicate filtering logic.